.github/workflows/support-window.yml

```diff
@@ -0,0 +1,55 @@
+name: Update support window diagram
+on:
+  # When translations are added/removed
+  pull_request:
+    branches: master
+    paths: README*.md
+  # Manually, when TypeScript is released
+  # https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
+  workflow_dispatch:
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          personal <- Sys.getenv("R_LIBS_USER")
+          dir.create(personal, recursive = TRUE)
+          install.packages(c("ggplot2", "svglite"), personal)
+        shell: Rscript {0}
+      - name: Fetch TypeScript versions and release dates from npm
+        run: |
+          npm view --json typescript time |
+            jq '
+              to_entries
+              # Earliest non-pre-release <major>.<minor>
+              | map(
+                (.key |= capture("^(?<version>[0-9]+\\.[0-9]+)\\.[0-9]+$").version)
+                | select(.key)
+              )
+              | unique_by(.key)
+              | from_entries
+            ' > docs/support-window.json
```

Comment:
Pulls the following table out of npm. If there's another, better source, please let me know.
```json
{
  "0.8": "2012-10-01T15:35:41.062Z",
  "0.9": "2013-06-18T18:06:35.222Z",
  "1.0": "2014-04-02T22:49:06.284Z",
  "1.3": "2014-11-12T15:06:54.123Z",
  "1.4": "2015-01-16T17:55:29.237Z",
  "1.5": "2015-07-20T14:47:42.460Z",
  "1.6": "2015-09-16T20:01:36.970Z",
  "1.7": "2015-11-30T20:37:03.883Z",
  "1.8": "2016-01-28T19:54:42.196Z",
  "1.9": "2016-04-28T20:04:35.616Z",
  "2.0": "2016-07-11T16:13:28.984Z",
  "2.1": "2016-11-08T17:31:39.827Z",
  "2.2": "2017-02-02T18:49:53.553Z",
  "2.3": "2017-04-10T16:03:31.897Z",
  "2.4": "2017-06-12T19:27:16.086Z",
  "2.5": "2017-08-17T17:52:57.895Z",
  "2.6": "2017-10-31T16:56:59.235Z",
  "2.7": "2018-01-31T17:22:12.576Z",
  "2.8": "2018-03-27T16:20:30.220Z",
  "2.9": "2018-05-31T16:14:25.828Z",
  "3.0": "2018-07-30T16:21:13.150Z",
  "3.1": "2018-09-27T14:31:16.554Z",
  "3.2": "2018-11-29T19:00:08.017Z",
  "3.3": "2019-01-31T19:57:04.866Z",
  "3.4": "2019-03-29T17:40:01.411Z",
  "3.5": "2019-05-29T16:24:22.635Z",
  "3.6": "2019-08-28T17:40:47.731Z",
  "3.7": "2019-11-05T16:11:44.457Z",
  "3.8": "2020-02-20T22:45:56.500Z",
  "3.9": "2020-05-12T21:13:54.150Z",
  "4.0": "2020-08-20T15:59:24.994Z",
  "4.1": "2020-11-19T21:54:03.703Z",
  "4.2": "2021-02-23T21:43:09.173Z",
  "4.3": "2021-05-26T17:25:54.805Z",
  "4.4": "2021-08-26T21:04:26.978Z"
}